code/builders/builder-vite/src/vite-config.ts

```diff
@@ -65,6 +65,7 @@ export async function commonConfig(
     base: './',
     plugins: await pluginConfig(options),
     resolve: {
+      conditions: ['storybook', 'stories', 'test'],
```

Comment:
Is this going to be merged correctly? Or does it override the user's base config? If so, we need to change this.

Reply:
We merge this with the user config using Vite's `mergeConfig`, so I think we're good.

https://vitejs.dev/guide/api-javascript#mergeconfig

https://github.com/storybookjs/storybook/blob/kasper/module-mocking/code/builders/builder-vite/src/vite-config.ts#L83

Reply:
Yes, I also manually tested that it doesn't remove base conditions like development and browser. Also the docs say:
> Vite has a list of "allowed conditions" and will match the first condition that is in the allowed list. The default allowed conditions are: import, module, browser, default, and production/development based on current mode. The resolve.conditions config option allows specifying *additional* allowed conditions.
https://vitejs.dev/config/shared-options#resolve-conditions

Reply:
Sounds good!